src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -726,10 +727,25 @@ def get_store_info(self) -> ServerModel:
 
         Returns:
             Information about the store.
+
+        Raises:
+            KeyError: If the deployment ID could not be loaded from the
+                database.
         """
         model = super().get_store_info()
         sql_url = make_url(self.config.url)
         model.database_type = ServerDatabaseType(sql_url.drivername)
+
+        # Fetch the deployment ID from the database and use it to replace the one
+        # fetched from the global configuration
+        with Session(self.engine) as session:
+            identity = session.exec(select(IdentitySchema)).first()
+
+            if identity is None:
+                raise KeyError(
+                    "The deployment ID could not be loaded from the database."
```

Comment:
As mention in the comment above, it would be good to see a 'way out' for the user in this case. Mayebe something like, please consider dropping the database or something